active/0003-extensible-resource-deployment-observability-improvements.md

```diff
@@ -0,0 +1,280 @@
+
+## Detailed design
+
+### API change for `GET /Microsoft.Resources/deployments/{deploymentId}`
+
+#### Adding `extensions`
+
+The `Microsoft.Resources/deployments` API will be updated to include an `extensions` array in the GET response body. This modification is intended to incorporate information about extensions that are involved in executing the parent deployment and nested deployments.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    ...
+    "providers": [
+      {
+        "namespace": "Microsoft.Storage",
+        "resourceTypes": [
+          {
+            "resourceType": "storageAccounts",
+            "locations": [
+              "eastus"
+            ]
+          }
+        ]
+      }
+    ],
++   "extensions": [
++     {
+        // deploymentId is used to uniquely identify an extension since a nested deployment could import the same extension with the same alias.
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "alias": "az",
++       "name": "AzureResourceManager",
++       "version": "1.0.0"
++     }  ,
++     {
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "alias": "kubernetes",
++       "name": "Kubernetes",
++       "version": "1.27.8",
++       "config": {
++         // Out of the scope of this REP, but it is required to enable Deployment Stacks integration.  
++       }
++     }
+    ]
+  }
+}
+```
+
+#### Updating the `outputResources` to include more identifiable information
+
+To improve the current structure where each `outputResources` object only has a resource `id`, it's essential to add more properties for identifying extensible resources, which lack resource IDs. These additions should include the resource symbolic name, resource type, resource API version, resource identifier, the resource ID of the deployment containing the resource, and the Bicep extension alias.
+
+```diff
+{
+  ...
+  "type": "Microsoft.Resources/deployments",
+  "properties": {
+    "outputResources": [
+      // Azure resource
+      {
+        "id": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Storage/storageAccounts/my-storage-account"
+      },
+      // Extensible resource
+      {
++       "deploymentId": "/subscriptions/00000000-0000-0000-0000-000000000001/resourceGroups/my-resource-group/providers/Microsoft.Resources/deployments/my-deployment",
++       "extension": "kubernetes"
++       "symbolicName": "myService",
++       "resourceType": "core/Service",
++       "apiVersion": "v1",
++       "identifiers": {
```

Comment:
I will disambiguate ARM vs EXT resources by the existence of the `id`\`identifiers` property just in case a future API update includes the other properties for Azure resources. Not sure if a `isExtensionResource` property is worth it.

Reply:
`extension` is required to uniquely identify extensible resources.